tests/Microsoft.Identity.Test.LabInfrastructure/UserQueryParameters.cs

```diff
@@ -39,5 +40,65 @@ public class UserQueryParameters
         public UserType? UserType { get; set; }
         public bool? IsExternalUser { get; set; }
         public B2CIdentityProvider? B2CIdentityProvider { get; set; }
+
+        public static UserQuery DefaultUserQuery => new UserQuery
+        {
+            IsMamUser = false,
+            IsMfaUser = false,
+            IsFederatedUser = false
+        };
+
+        public static UserQuery B2CLocalAccountUserQuery => new UserQuery
+        {
+            UserType = LabInfrastructure.UserType.B2C,
+            B2CIdentityProvider = LabInfrastructure.B2CIdentityProvider.Local
+        };
+
+        public static UserQuery B2CFacebookUserQuery => new UserQuery
+        {
+            UserType = LabInfrastructure.UserType.B2C,
+            B2CIdentityProvider = LabInfrastructure.B2CIdentityProvider.Facebook
+        };
+
+        public static UserQuery B2CGoogleUserQuery => new UserQuery
+        {
+            UserType = LabInfrastructure.UserType.B2C,
+            B2CIdentityProvider = LabInfrastructure.B2CIdentityProvider.Google
+        };
+
+        // generated code, re-generate or update manually if you change the members of this class !
+        #region Equals and GetHashCode
```

Comment:
You need to implement these so that a Dictionary that has UserQuery as key functions as expected. I miss Java + Lombok (AOP) that would just generate this on the fly ...